cmd/juju/action/common.go

```diff
@@ -197,18 +220,26 @@ func (c *runCommandBase) waitForTasks(ctx *cmd.Context, tasks []enqueuedAction,
 		}
 		if err != nil {
 			if errors.IsTimeout(err) {
-				return c.handleTimeout(tasks, resultReceivers)
+				return nil, c.handleTimeout(tasks, resultReceivers)
 			}
-			return errors.Trace(err)
+			return nil, errors.Trace(err)
 		}
 
 		resultReceivers.Add(result.receiver)
-		d := formatActionResult(result.task, actionResult, c.utc)
-		d["id"] = result.task // Action ID is required in case we timed out.
-		info[result.receiverId()] = d
+		resultData, resultExitCode := formatActionResult(result.task, actionResult, c.utc)
+		resultData["id"] = result.task // Action ID is required in case we timed out.
+		info[result.receiverId()] = resultData
+
+		// If any of the actions have a error exit code, then inform the user
+		// that their exec failed.
+		if resultExitCode < 1 {
```

Comment:
Why not `== 0` ?

Reply:
Result exit code is optional, so we can return `-1`.

Reply:
Ah ok then; thought it would just be zero if not tracked.